src/main/java/amazin/controller/BookController.java

```diff
@@ -97,48 +97,29 @@ public String deleteBook(@PathVariable("id") Long id, RedirectAttributes attribu
         return createRedirectViewPath(REQUEST_MAPPING_BOOK);
     }
 
-    @GetMapping("/fuzzySearchName")
-    public String fuzzySearchByBookName(@RequestParam(value = "keywords", required = false) String text, Model model) {
-        List<Book> searchResults = null;
-        try {
-            searchResults = searchservice.fuzzySearch(text, "name");
+    @GetMapping("/searchByName")
+    public String searchByBookName(@RequestParam(value = "keywords", required = true) String text,
```

Comment:
The user should not need to specify the type of search, it should be fuzzy by default and use wildcards if they use wildcards in their string.

Reply:
i have added a radio button where the use can choose what they want, this will not be entered manually

Reply:
I know but the user doesn't need to choose the algorithm they are using, you need to choose it based on their search.

Reply:
users don't know about wildcards to begin with, they will not know about putting * or anything like that. The default is fuzzy, also we can't assume that a book won't have * in it, books can be called anything.

Reply:
Then why do you have wildcard search?

Reply:
it's like advanced searches, they don't actually have to enter anything differently, for wildcard or anything else, they enter whatever they want and they get different results based on the search algorithm that they use

Reply:
I looked at all the code, I know it is, but I think its better for that choice to be made automatically not explicitly based on their search string.

Reply:
It is much easier to let the user choose the search algorithm they want, as it is difficult to predict what they want (I'm not working at amazon lol)

Reply:
You should run the three algorithms and return:
- the result of the first algorithm right outer joined with algorithm 2 then right outer joined with algorithm 3

Reply:
If instead, you want a more conservative search.

It would be smart to do a inner left join out of the three algorithms

Reply:
this is becoming more complicated than it has to be, I will instead use the fuzzy search which is the best for most cases